opkg: add opkg execution module

adelcast opened the pull request:
This change adds a new pkg module for the embedded package manager opkg (http://wiki.openwrt.org/doc/techref/opkg) which is used by embedded distributions, such as OpenWRT and OpenEmbedded. 

I enabled the module only for the NILinuxRT distribution, but it would be trivial to add support for others, like OpenWRT. Another (coming) pull request adds Salt support for NILinuxRT.

Signed-off-by: Alejandro del Castillo alejandro.delcastillo@ni.com

Commit:
opkg: add opkg execution module

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>

jfindlay commented on salt/modules/opkg.py:
```diff
@@ -0,0 +1,975 @@
+    return latest_version(name) != ''
+
+
+def version_cmp(pkg1, pkg2):
```
Rather than a `try..except Exception` could you rewrite this function:
```python
import salt.utils.decorators.which as _which
...
@which('opkg-compare-versions')
def version_cmp(pkg1, pkg2):
```

jfindlay commented on salt/modules/opkg.py:
```diff
@@ -0,0 +1,975 @@
+# -*- coding: utf-8 -*-
+'''
+Support for Opkg
+
```
Will you add a `.. versionadded:: Boron` here?  Thanks.

jfindlay commented:
@adelcast, very nice, thanks.

adelcast commented:
I made a change, squash my commits to have a clean history, then pushed. But apparently, a couple of commits were picked up too (shows on the UI).

my change is only on opkg.py...not sure why the other changes were picked up too

cachedout commented:
@adelcast Yeah, it looks like some systemd changes made it in to this PR. Let's hold off on merging this until tests finish and you can git rebase this commit to just include what's necessary. If you get stuck with that, feel free to close this one and re-open another. That's totally fine. :]

adelcast commented:
ok, I fixed the issues pointed out by jfindlay as well as some of python3 problem pointed by the linter (my pylint was not loading the python3 checks).

I rebased and pushed again, which unfortunately doesn't show diff's between submissions. Next time, I'll push incremental fixes instead of rebases.

jfindlay commented:
@adelcast, a rebase is fine with me, thanks for your work on this.

cachedout commented:
Great! Thanks @adelcast
